runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java

```diff
@@ -308,8 +314,15 @@ public Object deserialize(ByteBuf b, CorfuRuntime rt) {
             Record record = Record.parseFrom(data);
             Any payload = record.getPayload();
 
+            String fullMessageName = getFullMessageName(payload);
+            if (!messagesFdProtoNameMap.containsKey(fullMessageName)) {
+                log.error("messagesFdProtoNameMap doesn't contain the message type {} from payload {}",
```

Comment:
Do we want to explicitly ask, "Was this table opened with the correct openTable() command and appropriate schema?"

Reply:
Done.